composer.json

```diff
@@ -54,8 +55,9 @@
 	"scripts": {
 		"test": [
 			"composer validate --no-interaction",
-			"phpcs -p -s",
-			"phpunit"
+			"vendor/bin/phpcs",
+			"vendor/bin/covers-validator",
+			"vendor/bin/phpunit"
```

Comment:
Why was the removal of the `vendor/bin/` paths reverted? We just removed them everywhere because it appears they don't do anything (Composer does find the binaries without hard-coding the paths), and paths with slashes are not Windows-compatible. It doesn't feel productive if different developers change details like this back and forth without sharing their reasoning.

Reply:
Oh no... don't remove them!

If you remove them and you have a random phpunit version installed, composer will end up using that, rather than using the expected one that comes with the project.

Reply:
Are you sure? I just tested it and that isn't what happens.

I installed phpunit 5.x to /usr/bin/phpunit and ran "composer test" in a project that uses phpunit 4.x, and composer correctly used the proper phpunit even though only a plain "phpunit" was specified in composer.json.

And the documentation (<https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands>) seems to confirm that /vendor/bin is prepended to the path, so the local binaries should take precedence.

Reply:
Really? That's interesting! I can't easily test this as I don't have PHP or PHPUnit installed so will just take your word for it. The behavior definitely used to be different 3 or 4 years ago.

In that case it's not important to add vendor/bin/, though the explicitness still won't hurt.

Reply:
Thiemo said the paths are wrong on Windows, so it apparently does hurt. (I don't have any Windows machines to verify).